service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -861,14 +862,26 @@ public void editSiteEntry(JWTData userData, int entryId, UpdateSiteRequest editS
 
     int siteId = siteEntriesRecord.getSiteId();
 
-    db.transaction(configuration -> {
-      siteEntriesRecord.store();
-      // force unadopt only if we change the latest site entry of an adopted site to have no tree
-      if (!editSiteEntryRequest.isTreePresent()
-          && isAlreadyAdopted(siteId)
-          && entryId == latestSiteEntry(siteId).getId()) {
-        forceUnadoptSite(userData, siteId);
-      }
-    });
+    db.transaction(
+        configuration -> {
+          siteEntriesRecord.store();
+          // force unadopt only if we change the latest site entry of an adopted site to have no
+          // tree
+          if (!editSiteEntryRequest.isTreePresent()
+              && isAlreadyAdopted(siteId)
+              && entryId == latestSiteEntry(siteId).getId()) {
+            forceUnadoptSite(userData, siteId);
+          }
+        });
+  }
+
+  @Override
+  public void approveSiteImage(JWTData userData, int imageID) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+    checkImageExists(imageID);
+    SiteImagesRecord imageRecord =
+        db.selectFrom(SITE_IMAGES).where(SITE_IMAGES.ID.eq(imageID)).fetchOne();
+    imageRecord.setApprovalStatus(ImageApprovalStatus.APPROVED.getApprovalStatus());
+    imageRecord.store();
```

Comment:
Does this just store/push these changes to the database?

Reply:
Yes